vignettes/grouping.Rmd

```diff
@@ -0,0 +1,172 @@
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The 1.3.0 release of yardstick introduced an implementation for _group-wise metrics_. The use case motivating the implementation of this functionality is _fairness metrics_, though group-wise metrics have applications beyond that domain. Fairness metrics quantify the degree of disparity in a metric value across groups. To learn more about carrying out fairness-oriented analyses with tidymodels, see the blog post on the tidymodels website. This vignette will instead focus on group-wise metrics generally, clarifying the meaning of "group-wise" and demonstrating functionality with an example dataset.
+
+<!-- TODO: link to forthcoming tidymodels blog post -->
+
+```{r pkgs, message = FALSE}
+library(yardstick)
+library(dplyr)
+
+data("hpc_cv")
+```
+
+# Group-awareness
+
+Even before the implementation of group-wise metrics, _all_ yardstick metrics had been _group-aware_. When grouped data is passed to a group-aware metric, it will return metric values calculated for each group.
+
+To demonstrate, we'll make use of the `hpc_cv` data set, containing class probabilities and class predictions for a linear discriminant analysis fit to the HPC data set of Kuhn and Johnson (2013). The model is evaluated against a 10-fold cross-validation, and the predictions for all folds are included. 
+
+```{r hpc-cv}
+tibble(hpc_cv)
+```
+
+For the purposes of this vignette, we'll also add a column `batch` to the data and select off the columns for the class probabilities, which we don't need.
+
+```{r hpc-modify}
+set.seed(1)
+
+hpc <-
+  tibble(hpc_cv) %>%
+  mutate(batch = sample(c("a", "b"), nrow(.), replace = TRUE)) %>%
+  select(-c(VF, F, M, L))
+
+hpc
+```
+
+If we wanted to compute the accuracy of the first resampled model, we could write:
+
+```{r acc-1}
+hpc %>% 
+  filter(Resample == "Fold01") %>%
+  accuracy(obs, pred)
+```
+
+The metric function returns one row, giving the `.metric`, `.estimator`, and `.estimate` for the whole data set it is passed.
+
+If we instead group the data by fold, metric functions like `accuracy` will know to compute values for each group; in the input, each row will correspond to a Resample.
+
+```{r hpc-cv-2}
+hpc %>% 
+  group_by(Resample) %>%
+  accuracy(obs, pred)
+```
+
+Note that the first row, corresponding to `Resample`, gives the same value as manually filtering for the observations corresponding to the first resample and then computing the accuracy.
+
+This behavior is what we mean by group-awareness. When grouped data is passed to group-aware metric functions, they will return values for each group.
+
+# Group-wise metrics
+
+Group-wise metrics are associated with a data-column such that, when passed data with that column, the metric will temporarily additionally group by that column, compute values for each of the groups defined by the column, and then aggregate the values computed for the temporary grouping back to the level of the inputted data's grouping.
```

Comment:
```suggestion
Group-wise metrics are associated with a data-column such that, when passed data with that column, the metric will temporarily group by that column, compute values for each of the groups defined by the column, and then aggregate the values computed for the temporary grouping back to the level of the input data's grouping.
```

You address being group aware at the end very clearly, and I think "additionally" is kind of confusing here.